File: A_Gennady_and_a_Card_Game.py
t = input()
arr = list(map(str, input().split()))

f = 0

for i in arr:
    for j in range(len(i)):
        if (i[j] == t[j]):
            f = 1
            
            
res = "YES" if f == 1 else "NO"

print(res)

File: A_Gennady_and_a_Card_Game.java

import java.util.*;

public class A_Gennady_and_a_Card_Game {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String t = sc.nextLine();
        String[] list = sc.nextLine().split(" ");
        int f = 0;
        for(String card : list ) {
            for (int i = 0; i < card.length(); i++) {
                if (card.charAt(i) == t.charAt(i)) {
                    f = 1;
                    break;
                }
            }
        }
        String res = (f == 1) ? "YES" : "NO";
        System.out.println(res);
    }
}